### PR TITLE
Remove uneeded deps

### PR DESCRIPTION
**Pull request by paschdan**

### CODE REVIEW

File: Services/Xml/Client.php [hunk 1]
```diff
@@ -23,19 +26,17 @@
...
      * @var LoggerInterface
      */
     protected $logger;
-
+    /**
+     * @var ClientInterface
+     */
+    private $client;
     /**
      * @var SerializerInterface
```
Comment (staegi) [inline Services/Xml/Client.php]: No interface needed

File: Tests/Services/Xml/ClientTest.php [hunk 2]
```diff
@@ -201,58 +209,50 @@ public function testUpdateSoldItemsSuccess() {
...
+        $clientReflection = new \ReflectionProperty(Client::class, 'client');
+        $clientReflection->setAccessible(true);
+        $clientReflection->setValue($this->client, $guzzle);
     }
 
     /**
      * @param string $pathToXmlFile
      */
-    private function setMockHandlerForErrorTest($pathToXmlFile) {
+    private function setMockHandlerForErrorTest($pathToXmlFile)
```
Comment (staegi) [inline Tests/Services/Xml/ClientTest.php]: please use data provider for Response body

File: Services/Xml/Client.php [hunk 3]
```diff
@@ -150,6 +114,20 @@ public function updateSoldItems(array $orders, $detailLevel = AfterbuyGlobal::DE
     }
 
     /**
+     * Logs to a logger, when given
+     *
+     * @param string|LogLevel  $level
+     * @param string $message
+     * @param array  $context
```
Comment (annikaflemming) [inline Services/Xml/Client.php]: please align the parameters, also don't add two spaces before $level

File: Services/Xml/Client.php [hunk 4]
```diff
@@ -150,6 +114,20 @@ public function updateSoldItems(array $orders, $detailLevel = AfterbuyGlobal::DE
...
+     */
+    private function log($level, $message, array $context = array())
```
Comment (annikaflemming) [inline Services/Xml/Client.php]: please use [] instead of array()

File: Tests/Services/Xml/ClientTest.php [hunk 5]
```diff
@@ -201,58 +209,50 @@ public function testUpdateSoldItemsSuccess() {
...
-
-    /**
      * @param int $statusCode
      */
-    private function setMockHandlerForUnavailabilityTest($statusCode) {
+    private function setMockHandlerForUnavailabilityTest($statusCode)
+    {
         $mock = new MockHandler([
             new Response($statusCode)
         ]);
```
Comment (annikaflemming) [inline Tests/Services/Xml/ClientTest.php]: not your fault, but please make a oneliner of this

**Pull request closed without merging**